### ISSUE TITLE
Expand nbsite's test project

### ISSUE THREAD

**Issue by ceball** [pyviz/nbsite#71]
Although some of nbsite's testing comes via the projects using it (unfortunately), there are actually some internal tests (an example/test project). From TODOs in nbsite's tox.ini, the current test project doesn't yet cover:
* gallery
* reference manual

Note: the tests currently only check that everything ran ok. Still need to test the generated website turned out ok...

**Comment by jsignell** [comment]
As of https://github.com/pyviz/nbsite/pull/88 there is also some structure for writing unit tests. I think more tests of various permutations of generate_rst would be valuable there.

**Comment by ceball** [comment]
The unit tests are a great idea and were sorely missing.

The testing I was referring to here is from the other end, i.e. that nbsite can successfully build a whole website including reference manual, gallery, etc.

(Also useful as an example for someone to start from for their own project.)

**Comment by jsignell** [comment]
I agree. But I think the unit tests can test all the functionality in parts and then it'll just work as a whole.

**Comment by jbednar** [comment]
Just working because the parts work might be true, though I've never seen it actually be true in practice!  But here we'll have lots of deployed sites testing it as a whole, though perhaps not with immediate feedback...
